.gitlab-ci.yml

```diff
@@ -1099,25 +1060,39 @@ iot_agent_suse-x64:
   stage: package_build
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/suse_x64:$DATADOG_AGENT_BUILDERS
   tags: [ "runner:main", "size:2xlarge" ]
-  needs: ["build_iot_agent-deb_x64"]
-  variables:
-    AWS_CONTAINER_CREDENTIALS_RELATIVE_URI: /credentials
+  needs: [ "fail_on_non_triggered_tag"]
   before_script:
-    - source /root/.bashrc && conda activate ddpy3
+    # This is done inside entrypoint.sh
+    # - source /root/.bashrc && conda activate ddpy3
     - inv -e deps --verbose --dep-vendor-only --no-checks
   <<: *skip_when_unwanted_on_7
   script:
+    - echo "About to build iot agent for $RELEASE_VERSION_7"
+
+    - echo "Detected host architecture $(uname -m)"
+    # $DD_TARGET_ARCH is only set by Arm build images, so assume amd64 if not present
+    - echo "Target architecture ${DD_TARGET_ARCH:=amd64}"
+
     # remove artifacts from previous pipelines that may come from the cache
     - rm -rf $OMNIBUS_PACKAGE_DIR_SUSE/*
     # Artifacts and cache must live within project directory but we run omnibus in a neutral directory.
     # Thus, we move the artifacts at the end in a gitlab-friendly dir.
+    # Use --skip-deps since the deps are installed by `before_script`.
```

Comment:
I guess this line should be lower? The omnibus build command that uses `--skip-deps` is a few lines below